optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/routing/GraphHopperRouter.java

```diff
@@ -72,7 +73,7 @@ public long travelTimeMillis(Coordinates from, Coordinates to) {
         GHResponse ghResponse = graphHopper.route(ghRequest);
         // TODO return wrapper that can hold both the result and error explanation instead of throwing exception
         if (ghResponse.hasErrors()) {
-            throw new RuntimeException("No route", ghResponse.getErrors().get(0));
+            throw new DistanceCalculationException("No route", ghResponse.getErrors().get(0));
```

Comment:
Personally, I don't see the need to create a new exception type which doesn't provide anything extra on top of `RuntimeException` - except that it perhaps should've been `IllegalStateException`. A more descriptive message avoids the need for a new exception type.

Reply:
1. Generic exceptions shouldn't be thrown. SonarCloud [warns about that](https://sonarcloud.io/project/issues?id=org.optaweb.vehiclerouting%3Aoptaweb-vehicle-routing&issues=AW5u71hIs0lcjfPwd2a8&open=AW5u71hIs0lcjfPwd2a8).
2. Throwing a more specialized exception type makes testing more precise.
3. I'm not against reusing existing Java exceptions but IllegalStateException's [description](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalStateException.html) doesn't fit this situation.

Reply:
I agree it shouldn't have been a `RuntimeException`, as that is indeed generic. If not `IllegalStateException` either, then for sure `IllegalArgumentException`, as that's what it is - an illegal combination of arguments.

Reply:
IllegalArgumentException: Thrown to indicate that a method has been passed an illegal or inappropriate argument.

I don't think that applies either. `[100,0]` might be considered an illegal argument as a latitude over 90° is not a valid coordinate.

Anyway, a runtime exception indicates a programming error. As a programmer you have no way to check whether there is a route between `from` and `to` coordinates other than calling the method, so I actually think that a better approach would be to return a response object that holds either a value or an error.

Reply:
Yes, some kind of an `Either` construct would be ideal - I was assuming we don't have that.

Reply:
We don't but it's emerging... https://www.baeldung.com/vavr-try.

Reply:
I was using that for a while. Not a bad library, but I found that adding a 1MB JAR for 1 class just wasn't worth it for me.